uiServices/services.go

```diff
@@ -3,54 +3,61 @@ package services
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"github.com/Sirupsen/logrus"
+	"github.com/xeipuuv/gojsonschema"
 	"github.com/xtracdev/automated-perf-test/perfTestUtils"
 	"net/http"
 	"os"
 	"strings"
-	"fmt"
-	"github.com/xeipuuv/gojsonschema"
 )
 
 func configsHandler(rw http.ResponseWriter, req *http.Request) {
+	configPathDir := req.Header.Get("configPathDir")
 
-		configPathDir := req.Header.Get("configPathDir")
-		//error check to ensure file path ends with "\"
-		if !strings.HasSuffix(configPathDir, "/") {
-			configPathDir = configPathDir + "/"
-		}
+	config := perfTestUtils.Config{}
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(req.Body)
 
+	defer req.Body.Close()
+	err := json.Unmarshal(buf.Bytes(), &config)
 
-		config := perfTestUtils.Config{}
-		buf := new(bytes.Buffer)
-		buf.ReadFrom(req.Body)
-		defer req.Body.Close()
+	if len(configPathDir) < 1 {
+		logrus.Error("File path is length too short", err)
+		rw.WriteHeader(http.StatusBadRequest)
+		return
 
-		err := json.Unmarshal(buf.Bytes(), &config)
+	}
+	//error check to ensure file path ends with "\"
+	if !strings.HasSuffix(configPathDir, "/") {
+		configPathDir = configPathDir + "/"
+	}
 
 	if err != nil {
 		logrus.Error("Failed to unmarshall json body", err)
 		rw.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	if FilePathExist(configPathDir) == false {
+	if !FilePathExist(configPathDir) {
 		logrus.Error("File path does not exist", err)
-		rw.WriteHeader(http.StatusNotFound)
+		rw.WriteHeader(http.StatusBadRequest)
 		return
 
 	}
-
-	if validdateJsonWithSchema(buf.Bytes()) == true {
+	if validateJsonWithSchema(buf.Bytes()) {
```

Comment:
Couple of points:

1. Go-style recommends that you handle errors in the if statements, and leave the happy path outside the conditions (below for example, illustrates this style).
2. You also need to handle returning a 400 when the Json schema is not valid.
3. move the call to validateJsonWithSchema(buf.Bytes()) above the unmarshalling because this schema is our contract. we want to fail with the header and the schema validation before we fail for unmarshalling.



```
if !validateJsonWithSchema(buf.Bytes()) {

  rw.WriteHeader(http.StatusBadRequest)
  return
}

if !writerXml(config, configPathDir) {
  rw.WriteHeader(http.StatusInternalServerError)
  return
}
  		
rw.WriteHeader(http.StatusCreated)
```